metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -794,11 +794,20 @@ def _generate_single_profile(
             **kwargs,
         }
 
-        if self.config.bigquery_temp_table_schema is not None:
-            bigquery_temp_table = (
-                f"{self.config.bigquery_temp_table_schema}.ge-temp-{uuid.uuid4()}"
-            )
-            ge_config["bigquery_temp_table"] = bigquery_temp_table
+        if custom_sql or self.config.limit or self.config.offset:
```

Comment:
Can you add comments about when and why the temp table is being created? The logic has a significant change from the earlier code.